src/AutoRest.CSharp/MgmtTest/Generation/ResourceTestWriter.cs

```diff
@@ -0,0 +1,214 @@
+                        _writer.Append($"var createOperation = ");
+                        _collectionTestWriter!.WriteInvokeExampleInstanceMethod(_resource.ResourceCollection!.CreateOperation!, async, "CreateOrUpdate", exampleGroup.Examples.First());
+                        var respType = GenResponseType(_resource.ResourceCollection!.CreateOperation!, true, "CreateOrUpdate");
+                        var resultTypeDataName = _resource.ResourceCollection!.CreateOperation!.IsLongRunningOperation() ? ((Mgmt.Output.MgmtLongRunningOperation)respType.Arguments[0].Implementation).ResultType?.Name : ((Mgmt.Output.NonLongRunningOperation)respType.Arguments[0].Implementation).ResultDataType?.Name;
+                        if (resultTypeDataName != $"{_resource.Type.Name}Data" && _resource.ResourceCollection!.GetOperation!.ReturnType?.Name == $"{_resource.Type.Name}Data") {
+                            _writer.Line($"return collection.Get(createOperation.Value.Name);");
+                        }
+                        else {
+                            _writer.Line($"return createOperation.Value;");
+                        }
+                        break;
+                    }
+                }
+                else
+                {
+                    throw new Exception("// TODO: create/get resource when there is no create example");
+                }
+            }
+        }
+
+        public void WriteGetResource(Resource resource, bool async)
+        {
+            var resourceVariableName = useVariableName(_resource.Type.Name.FirstCharToLowerCase());
+            _writer.Line($"var {resourceVariableName} = {GetAwait(async)} Get{_resource.Type.Name}{GetAsyncSuffix(async)}();");
+            if (resource != _resource) {
+                var childResourceVariableName = useVariableName(resource.Type.Name.FirstCharToLowerCase());
+                _writer.Line($"var {childResourceVariableName} = {resourceVariableName}.Get{resource.Type.Name}();");
+            }
+        }
+
+        protected override CSharpType? WrapResourceDataType(CSharpType? type, MgmtRestOperation operation)
+        {
+            if (!IsResourceDataType(type, operation))
+                return type;
+
+            return _resource.Type;
+        }
+
+        protected override bool IsResourceDataType(CSharpType? type, MgmtRestOperation operation)
+        {
+            return _resource.Type.Equals(type);
+        }
+
+        protected void WriteMethodTest(Resource resource, MgmtClientOperation clientOperation, bool async)
+        {
+            Debug.Assert(clientOperation != null);
+            var methodName = clientOperation.Name;
+
+            BuildParameters(clientOperation, out var operationMappings, out var parameterMappings, out var methodParameters);
+            foreach ((var branch, var operation) in getSortedOperationMappings(clientOperation))
```

Comment:
Instead of using this `getSortedOperationMappings`, I recommend we override the `BuildParameters` method here, to sort the out variables correspondingly

Reply:
And for those variables that are not actually used, we can use the discard identifier to discard it explicitly.
```suggestion
            BuildParameters(clientOperation, out var operationMappings, out _, out var methodParameters);
            foreach ((var branch, var operation) in getSortedOperationMappings(clientOperation))
```

Reply:
Also please feel free to change the interface in `MgmtBaseClientWriter` if you feel some interface is not reasonable to use.